### PR TITLE
chore(ui): Node CVE single page tests

### PR DESCRIPTION
**Pull request by dvail**
### Description

Adds tests for the Node CVE single page based on common abstractions merged for the Node CVE overview page. Tests are roughly one test addition per commit in this PR.

### User-facing documentation

- [ ] CHANGELOG is updated **OR** update is not needed
- [ ] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing

- [ ] inspected CI results

#### Automated testing

- [ ] added e2e tests

#### How I validated my change

Local cypress run, CI

### CODE REVIEW

File: ui/apps/platform/cypress/fixtures/vulnerabilities/nodeCves/getAffectedNodes.json
Comment (dvail) [inline ui/apps/platform/cypress/fixtures/vulnerabilities/nodeCves/getAffectedNodes.json:1]: These and other changes in fixtures are intended to make it even more obvious when viewing test recordings that the data in use is mocked, and not returned from central.

File: ui/apps/platform/cypress/helpers/request.js
Comment (dvail) [inline ui/apps/platform/cypress/helpers/request.js:1]: Little follow up refactor RE: https://github.com/stackrox/stackrox/pull/12133#discussion_r1695429078

File: ui/apps/platform/cypress/integration/vulnerabilities/nodeCves/cveDetailPage.test.ts [hunk 1]
```diff
@@ -65,67 +94,239 @@ describe('Node CVEs - CVE Detail Page', () => {
...
+                expectRequestedSort({ field: 'Operating System', reversed: false })
+            );
+
+            // check that the Affected components column is not sortable
+            queryTableHeader('Affected components');
+            queryTableSortHeader('Affected components').should('not.exist');
+        });
     });
 
     it('should filter the Node table', () => {
```
Comment (dvail) [inline ui/apps/platform/cypress/integration/vulnerabilities/nodeCves/cveDetailPage.test.ts:209]: Given that the data is mocked, there isn't much use for many of the tests described in the original comment block. Similarly, there are no per-row assertions done on the mocked data.

File: ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/NodeCve/NodeCvePage.tsx [hunk 2]
```diff
@@ -137,6 +137,7 @@ function NodeCvePage() {
                     searchFilterConfig={searchFilterConfig}
                     onFilterChange={(newFilter, searchPayload) => {
                         setSearchFilter(newFilter);
+                        setPage(1, 'replace');
```
Comment (dvail) [inline ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/NodeCve/NodeCvePage.tsx:140]: Another subtle bug discovered during test writing...

**Comment by dvail** [comment]
ocp-4-12 failure is due to a failure to create the cluster

**Review by sachaudh — APPROVED** [review]

**Pull request merged**